driver/pdist_driver.hpp

```diff
@@ -0,0 +1,398 @@
+    inflags.AddInputFlag("contiguous",
+                         'C',
+                         "1",
+                         "Tensor is contiguous or not (Default=1 for contiguous tensor)",
+                         "int");
+    inflags.AddInputFlag("power",
+                         'p',
+                         "2",
+                         "p value for the p-norm distance to calculate between each vector pair, "
+                         "value in range [0, inf] (Default=2)",
+                         "double");
+
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify Each Layer (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "0", "Time Each Layer (Default=0)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int PdistDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    is_contiguous = inflags.GetValueInt("contiguous") > 0 ? true : false;
+
+    p = inflags.GetValueDouble("power");
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int PdistDriver<Tgpu, Tref>::GetandSetData()
+{
+    auto in_dims         = inflags.GetValueTensor("dims").lengths;
+    auto in_strides      = ComputeStrides(in_dims);
+    auto N               = in_dims[0];
+    auto output_dim_size = N * (N - 1) / 2;
+
+    if(output_dim_size == 0)
+    {
+        MIOPEN_THROW("Invalid input tensor dimensions: MIOpen doesn't handle for empty tensors "
+                     "yet, hence input.shape[0] cannot be {0, 1}.");
+    }
+
+    std::vector<int> output_dims({output_dim_size});
+
+    if(SetTensorNd(inputDesc, in_dims, in_strides, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing input tensor: " + inflags.GetValueStr("dims") + ".");
+
+    if(SetTensorNd(outputDesc, output_dims, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing output tensor: {" + std::to_string(output_dim_size) + "}.");
+
+    if(SetTensorNd(doutputDesc, output_dims, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing doutput tensor: {" + std::to_string(output_dim_size) + "}.");
+
+    if(SetTensorNd(dinputDesc, in_dims, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing dinput tensor: " + inflags.GetValueStr("dims") + ".");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int PdistDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
+{
+    size_t input_size   = GetTensorSize(inputDesc);
+    size_t output_size  = GetTensorSize(outputDesc);
+    size_t doutput_size = GetTensorSize(doutputDesc);
+    size_t dinput_size  = GetTensorSize(dinputDesc);
```

Comment:
`dinput_size` is actually the same with `input_size`, might eliminate this if needed